ui/library/src/context/DocumentContext.ts

```diff
@@ -5,29 +5,63 @@ import { Dimensions } from '../components/types/boundingBox';
 import { OutlineNode } from '../components/types/outline';
 import { Nullable } from '../components/types/utils';
 import { logProviderWarning } from '../utils/provider';
+import { PageRotation } from '../utils/rotate';
+import { calculateTargetPosition } from '../utils/scroll';
+
+export type OutlinePosition = {
+  pageNumber: number;
+  dest: string;
+  leftPoint: number;
+  bottomPoint: number;
+};
+
+export type OutlinePositionsByPageNumberMap = Map<number, OutlinePosition[]>;
+
+export type OutlineTarget = {
+  dest: string;
+  leftPx: number;
+  topPx: number;
+};
 
 export interface IDocumentContext {
   numPages: number;
   outline: Nullable<Array<OutlineNode>>;
+  outlinePositions: Nullable<OutlinePositionsByPageNumberMap>;
   pageDimensions: Dimensions; // Scaled at 100%
   pdfDocProxy?: pdfjs.PDFDocumentProxy;
+  getOutlineTargets: (opts: {
+    pageNumber?: number;
+    pageIndex?: number;
+    scale: number;
+    rotation: PageRotation;
+    pageDimensions: Dimensions;
+  }) => OutlineTarget[];
```

Comment:
Why not creating a type for this? I feel like I've seen similar structures many times. It would be a good idea to give them types and organize them accordingly

Reply:
@yensung done